src/main/java/org/opensearch/security/rest/SecurityConfigUpdateAction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.security.rest;
+
+import com.google.common.collect.ImmutableList;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.rest.*;
+import org.opensearch.rest.action.RestActions.NodesResponseRestListener;
+import org.opensearch.security.action.configupdate.ConfigUpdateAction;
+import org.opensearch.security.action.configupdate.ConfigUpdateRequest;
+import org.opensearch.security.configuration.AdminDNs;
+import org.opensearch.security.ssl.transport.PrincipalExtractor;
+import org.opensearch.security.ssl.util.SSLRequestHelper;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.user.User;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class SecurityConfigUpdateAction extends BaseRestHandler {
+
+    private static final List<Route> routes = addRoutesPrefix(ImmutableList.of(
+            new Route(PUT, "/configupdate")),
```

Comment:
Why is it necessary to add a new endpoint? Can't we use the existing one `/securityconfig` here?

Reply:
This patch does not modify the basic working principle of securityadmin, it just moves the communication between securityyadmin and the backend from the transport protocol to the REST layer. 

`securityadmin` uses a number of different actions to work:

- `IndexAction` to update the security index. This is a standard OpenSearch action and is thus also available on the REST layer
- `ConfigUpdateAction` to notify the security plugin that the index contents have been changed and that the content needs to be reloaded. So far, it was only necessary as transport action, thus it was not exposed on the REST layer. Thus, we now need to expose it on the REST layer which is done by the class in question.
- `WhoAmIAction` to check authentication. This is also used by `securityadmin` for auxilliary tasks. This way also not exposed on REST before.

Using the existing `config` REST APIs won't work, as these only work on single config entries, not the whole config. Securityadmin on the other hand must be able to update the whole configuration of several config types at one time.

Of course, it would be nicer to implement all this using a brand-new dedicated REST API, and generally this would be the way I would recommend. However, this is not possible on the given time-frame.